# CA1 Data Visualization Higher Diplome Lvl8  

### Student - Linika Almeida, ID - 2023045

### Institution - CCT College

Description: This project is part of the HDIP Data Analytics course for the module of Data Visualization Techniques and implements a analyse based on a Board Game dataset.

### Importing Libraries

In [16]:
import warnings
warnings.filterwarnings('ignore')

#Import libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import scipy.stats as stats

In [17]:
#Open dataset
df = pd.read_csv('board_games.csv')

### Data Description

You have been retained by a retail company to analyse a dataset based on board games. This analysis will help determine the sales strategy for the company in their upcoming Winter season.   
  
The companies CTO also requires you to rationalize all the decisions that you have made in your report.   
  
This rationalization MUST include your visualization design decisions, how you have engineered the data, feature selection and any other information that you deem relevant.   

### Exploratry Data Analyse

In the EDA I will do some assumptions, do a comprehension of the dataset and check missing values

In [18]:
#Show data head
df.head()

game_id                                        description  \
0        1  Die Macher is a game about seven sequential po...   
1        2  Dragonmaster is a trick-taking card game based...   
2        3  Part of the Knizia tile-laying trilogy, Samura...   
3        4  When you see the triangular box and the luxuri...   
4        5  In Acquire, each player strategically invests ...   

                                          image  max_players  max_playtime  \
0   //cf.geekdo-images.com/images/pic159509.jpg            5           240   
1   //cf.geekdo-images.com/images/pic184174.jpg            4            30   
2  //cf.geekdo-images.com/images/pic3211873.jpg            4            60   
3   //cf.geekdo-images.com/images/pic285299.jpg            4            60   
4   //cf.geekdo-images.com/images/pic342163.jpg            6            90   

   min_age  min_players  min_playtime            name  playing_time  ...  \
0       14            3           240      Die Macher           240  ...   
1       12            3            30    Dragonmaster            30  ...   
2       10            2            30         Samurai            60  ...   
3       12            2            60  Tal der Könige            60  ...   
4       12            3            90         Acquire            90  ...   

                        artist                        category compilation  \
0          Marcus Gschwendtner  Economic,Negotiation,Political         NaN   
1                   Bob Pepper               Card Game,Fantasy         NaN   
2              Franz Vohwinkel      Abstract Strategy,Medieval         NaN   
3                          NaN                         Ancient         NaN   
4  Scott Okumura,Peter Whitley                        Economic         NaN   

                designer expansion  \
0     Karl-Heinz Schmiel       NaN   
1  G. W. "Jerry" D'Arcey       NaN   
2          Reiner Knizia       NaN   
3      Christian Beierer       NaN   
4            Sid Sackson       NaN   

                                              family  \
0         Country: Germany,Valley Games Classic Line   
1                                   Animals: Dragons   
2  Asian Theme,Country: Japan,Knizia tile-laying ...   
3             Country: Egypt,Promotional Board Games   
4                                3M Bookshelf Series   

                                            mechanic  \
0  Area Control / Area Influence,Auction/Bidding,...   
1                                       Trick-taking   
2  Area Control / Area Influence,Hand Management,...   
3  Action Point Allowance System,Area Control / A...   
4       Hand Management,Stock Holding,Tile Placement   

                                           publisher average_rating  \
0  Hans im Glück Verlags-GmbH,Moskito Spiele,Vall...        7.66508   
1                           E.S. Lowe,Milton Bradley        6.60815   
2  999 Games,ABACUSSPIELE,Astrel Games,Ceilikan J...        7.44119   
3                                             KOSMOS        6.60675   
4  3M,Avalon Hill,Avalon Hill (Hasbro),Dujardin,G...        7.35830   

  users_rated  
0        4498  
1         478  
2       12019  
3         314  
4       15195  

[5 rows x 22 columns]

Lets check some information in this dataset

In [19]:
#Show details about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10532 entries, 0 to 10531
Data columns (total 22 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   game_id         10532 non-null  int64  
 1   description     10532 non-null  object 
 2   image           10531 non-null  object 
 3   max_players     10532 non-null  int64  
 4   max_playtime    10532 non-null  int64  
 5   min_age         10532 non-null  int64  
 6   min_players     10532 non-null  int64  
 7   min_playtime    10532 non-null  int64  
 8   name            10532 non-null  object 
 9   playing_time    10532 non-null  int64  
 10  thumbnail       10531 non-null  object 
 11  year_published  10532 non-null  int64  
 12  artist          7759 non-null   object 
 13  category        10438 non-null  object 
 14  compilation     410 non-null    object 
 15  designer        10406 non-null  object 
 16  expansion       2752 non-null   object 
 17  family          7724 non-null  

From this output, we can see that the DataFrame has 10532 rows (entries) and 22 columns. The columns have different data types, including int64, float64, and object

Some features detected in data head are not relevant for the purpose of this analyse so they will be deleated, 
and they also contain NuN values. So for that reasons the follow feautures are going to be deleted: image, 
thumbnail, artist, compilation, expantion, publisher

In [20]:
# Drop columns that are not relevant for the analyse
df.drop(['image', 'thumbnail', 'artist', 'compilation', 'designer', 'expansion', 'family', 'publisher'], axis=1, inplace=True)

Checkinh the type of data we have in this dataset

In [21]:
#Describe the type of data
df.dtypes.value_counts()

int64      9
object     4
float64    1
dtype: int64

The dtype shows four features are object: ‘description’, ‘name’, ‘category’ and 'mechanic'. It is more appropriate convert them from an object to a category

In [22]:
# Convert object type for categorical features
df['description'] = df['description'].astype('category')
df['name'] = df['name'].astype('category')
df['category'] = df['category'].astype('category')
df['mechanic'] = df['mechanic'].astype('category')

#### Cleaning dataset

I am going to check the mising values in this dataset, once it can cause problem in the analyse, such a biased result, incorrect statistial inferences and error in machine learning models.

In [23]:
# Check missing values
df.isnull().sum()

game_id             0
description         0
max_players         0
max_playtime        0
min_age             0
min_players         0
min_playtime        0
name                0
playing_time        0
year_published      0
category           94
mechanic          950
average_rating      0
users_rated         0
dtype: int64

The features 'category' and 'mechanic' have missing values, both are categorical variables 

In [24]:
# Find rows with NaN values
rows_null_values = df[df.isnull().any(axis=1)]

# Display rows with NaN values
print(rows_null_values)

       game_id                                        description  \
23          25  In this fantasy sequel to Twilight Imperium, p...   
30          32  In Bison, a.k.a. Buffalo, one player represent...   
33          37  This odd little game put out by Games Workshop...   
34          38  Runes is a different kind of word game where p...   
35          39  Based on the books of Marion Zimmer Bradley th...   
...        ...                                                ...   
10496   206940  In Carcassonne: Amazonas, players sail their b...   
10509   208766  Description from the publisher:&#10;&#10;The c...   
10518   209886  Gambling continues in the glamorous world of L...   
10520   210342  Timeline: British History is a card game playe...   
10523   212382  In Yangtze, players are distributors on the Ya...   

       max_players  max_playtime  min_age  min_players  min_playtime  \
23               6           200       12            2           200   
30               2         

After analyse the rows with NaN values I decide to keep them and classify these line as special labels. Once the other features from the same line have important data for my analyse

In [25]:
# Specify the special label for NaN values
special_label = 'Special_Label'  # You can choose any label you prefer
special_label1 = 'Special_Label1'

# Add the special label as a new category
df['category'] = df['category'].cat.add_categories(special_label)
df['mechanic'] = df['mechanic'].cat.add_categories(special_label1)

# Replace NaN values with the special label
df['category'].fillna(special_label, inplace=True)
df['mechanic'].fillna(special_label1, inplace=True)

# Display the updated DataFrame
print(df)

       game_id                                        description  \
0            1  Die Macher is a game about seven sequential po...   
1            2  Dragonmaster is a trick-taking card game based...   
2            3  Part of the Knizia tile-laying trilogy, Samura...   
3            4  When you see the triangular box and the luxuri...   
4            5  In Acquire, each player strategically invests ...   
...        ...                                                ...   
10527   214996  Description from the publisher:&#10;&#10;Silve...   
10528   215437  Codex: Card-Time Strategy is a customizable, n...   
10529   215471  Time to walk about town and take some pictures...   
10530   216201  The race is on for the robots of the Robo Rall...   
10531   216725  The deluxe edition comes in a double tall box ...   

       max_players  max_playtime  min_age  min_players  min_playtime  \
0                5           240       14            3           240   
1                4         